src/types/button.ts

```diff
@@ -0,0 +1 @@
+export type ButtonKind = 'default' | 'primary' | 'danger' | 'primary--danger' | 'flat'
```

Comment:
nit: I'd keep to a single dash for `primary--danger`

```suggestion
export type ButtonKind = 'default' | 'primary' | 'danger' | 'primary-danger' | 'flat'
```

Also, can you give me an example where we would need "danger" vs "primary-danger"? It would be nice to not have two "danger" styles. But mostly having one two word kind is annoying me lol.

Reply:
Yeah, I feel the annoyance. A standard danger button is for any ol' destructive activity. So you might see one of these:
<img width="75" alt="image" src="https://github.com/PrefectHQ/prefect-design/assets/6776415/d83a7664-8a7e-49e9-9afa-41facf0e2e88">

Whereas a primary-danger button functions like a primary button, it has prominent visibility as it constitutes the "next step in a journey". So in a confirmation modal, etc. you'd have a primary-danger button.

<img width="117" alt="image" src="https://github.com/PrefectHQ/prefect-design/assets/6776415/b6935492-3fd1-4818-bff1-d3b64d54f96e">

Reply:
I think `dangerous` can actually be a prop that applies to any button type

Reply:
Ah so just `<p-button dangerous>` and `<p-button kind="primary" dangerous>`?

Reply:
right right, which also enables `<p-button icon="TrashIcon" kind="flat" dangerous />`, which I'll add styles for

Reply:
At that point I might go back to accepting just the names like `<p-button primary dangerous>`

Reply:
I like that. Either with or without the kind property.

Reply:
Yeah.. Only combo you wouldn't really want is `flat primary`.. will wrestle with that

Reply:
Agreed. But in that case I think primary just takes over. 

I've seen quite a few button/theme systems designed that way.